spec.go

```diff
@@ -14,7 +14,20 @@ const (
 
 // Interface describes the methods provided by the appr client.
 type Interface interface {
+	// DeleteRelease removes a release from the server.
+	DeleteRelease(ctx context.Context, name, release string) error
+	// GetReleaseVersion queries CNR for the release version of the chart
+	// represented by the given name and channel.
 	GetReleaseVersion(ctx context.Context, name, channel string) (string, error)
+	// PromoteChart puts a release of the given chart in a channel.
+	PromoteChart(ctx context.Context, name, release, channel string) error
```

Comment:
Do we still need `PromoteChart` BTW?

Reply:
I just added it for completeness. Overall we are moving away from CNR to use catalog repos since that is how the Helm community has gone. The new chart CRD just has a `tarballURL`.

I am adding `PullChartTarball` to `helmclient`. I think that's all we need from `apprclient` long term. Once chart-operator supports chart CRDs I suggest we add a deprecation notice to `apprclient`. WDYT?

Reply:
yeah, we shouldn't need anything from `apprclient` as we're fully relying on upstream helm client functionality, not sure if even `PullChartTarball` would be needed, as we are not using an appr to pull that tarball we are using helm repos and that functionality should be in normal helmclient already

Reply:
Yes I'm adding PullChartTarball to helmclient. The arguments are different because it just takes the tarball URL. Here there is magic to calculate the URL from the channel etc.